### ISSUE TITLE
Feature request: Ability to record postcards

### ISSUE THREAD

**Issue by fuecy** [monicahq/monica#5517]
The ability to record postcards from and to a person together with an event.
So for example, i would like to see (and register) everybody that i received a xmas card from in 2020.
Or everybody that send me a birthday card in 2021
Or everybody i send a xmas card to in 2020

Currently i make tags and tag the persons with the event but after a few years that doesnt really work as it becomes very clumsy.

I feel its a nice addition to monica.

**Comment by virtadpt** [comment]
Agreed.  It would also be very helpful for keeping track of postcard and letter exchange.

**Comment by mgrhm** [comment]
@fuecy: I’m not entirely sure what you’re requesting here.

Are you asking for the ability to make an Activities, say your birthday in 2021, then record against that event everyone who sent you a birthday card or present?  Do you want to also record the contents of those cards?  Where do you foresee that function being displayed?

I think as it stands, Activities already do a lot of heavy lifting and adding this too would overload them.  Events already have two mandatory pieces of information, and four optional ones:

<img width="527" alt="Screenshot 2021-09-19 at 17 22 52" src="https://user-images.githubusercontent.com/21129642/133935051-c2ea118c-7913-486f-9b89-f21e1f1d8b5e.png">

I think your feature request would be easier to understand if you could explain how you think this would be implemented and where – not the code implementation, but how you imagine the interface for doing it, the steps it would take to record this data, and how it would be displayed.

**Comment by mgrhm** [comment]
@virtadpt: What you’re asking for seems to be something different: a function or page for keeping track of letters to-and-from, and presumably holding copies of those letters too.  It doesn't appear to be linked to an Event or Activity like @fuecy’s request.

I see what you’re asking for as being a separate page – called, say, “Correspondence Tracker” or “Correspondence Hub” – which could be laid out something like this:

> **19 September 2021:** [Letter to Bob]() (no reply received)
> **11 September 2021:** [Letter from Bob]() (reply sent [19/09/2021]())
> **04 August 2021:** [Letter to Bob]() (reply received [11/09/2021]())
> **03 June 2021:** [Letter from Samira]() **(no reply sent)**

It would also have sections for tracking letters you need to reply to, and ones you’re awaiting a reply from, and a checkbox to mark either as no-reply-needed.  Frequent correspondents, and the ability to filter it to just letters to/from Bob.  It would also be handy to track letters to people and organisations that *aren’t* contacts – like complaint letters to your bank, for example.  I’m sure there’s a bunch of other stuff I’m missing.

I currently track letters sent and received in Conversations, along with any other correspondence such as cards of various kinds and emails.  It’s [not perfect](https://github.com/monicahq/monica/issues/4801), but it works for now.  Something like what you’re suggesting would be really useful though

**Comment by fuecy** [comment]
@mgrhm Maybe activities does indeed much of what i was looking for. I wasn't aware the capability of linking other persons to the activity. But in addition, i would like to specify the relationship to the activity, so a person has a many-to-many relation to an activity where the relationship itself also contains a type. Then i can distinct between who was invited to a party, and who actively joined the party. Invited and joined could be different type.
It would be nice to have a list of activities where you can open an activity and see the relations. Now the only way to see (or search activities) is to either look into the diary or you know a person has joined.

Just to come back to my initial post, i have 2 usecases which can be solved with this.
1. I would like to record to who i send xmas cards to last  year, from who i received xmas cards. Or basically any card, could be a birtday or marriage. This indeed cloud be recorded as an event type, if i cloud specify using the relationship type if its an send or received letter, this will work.
2. I would like to record who of my kids joined my boys kids party last year, who did i invite and who did actually come and what present they brought. Personally, i would not even be there myself but my wife. If can make an activity and specify the relationships to the persons, that would be great. For example, i can also specify who of the other parents helped managing the 10 kids.

Regarding the interface:
- In the header bar, i would like to click on activities, to open a page that lists all activities or can search/filter on activity dates or names.
- A page that shows an individual activity that shows all the relationships to that activity and an description.
- A section to the Personalisation page to maintain the activity relationship types.
- Nice to have: A relation between the presents and activities
- Nice to have: A relation between depts and activities
